### PR TITLE
Minimal changes to jacobian in calculus.jl to allow for nonsquare Jacobians

### PR DESCRIPTION
**Pull request by gaperez64**
I enabled the CI workflow on my clone and things seemed OK, I hope there are some tests being run there becase I did not run any checks/tests manually :D

### CODE REVIEW

**Comment by gaperez64** [comment]
oh and this resolves #371

File: src/calculus.jl [hunk 1]
```diff
@@ -252,10 +252,9 @@ evaluated at the vector `vals`. If `vals` is omitted, it is evaluated at zero.
 """
 function jacobian(vf::Array{TaylorN{T},1}) where {T<:Number}
     numVars = get_numvars()
-    @assert length(vf) == numVars
-    jac = Array{T}(undef, numVars, numVars)
+    jac = Array{T}(undef, numVars, length(vf))
 
-    @inbounds for comp = 1:numVars
+    @inbounds for comp = 1:length(vf)
```
Comment (lbenet) [inline src/calculus.jl:257]: I think the following is recommended (it should have been changed before, but ...):
```suggestion
    @inbounds for comp in eachindex(vf)
```

File: src/calculus.jl [hunk 2]
```diff
@@ -264,10 +263,10 @@ end
 function jacobian(vf::Array{TaylorN{T},1}, vals::Array{S,1}) where {T<:Number,S<:Number}
     R = promote_type(T,S)
     numVars = get_numvars()
-    @assert length(vf) == numVars == length(vals)
-    jac = Array{R}(undef, numVars, numVars)
+    @assert numVars == length(vals)
+    jac = Array{R}(undef, numVars, length(vf))
 
-    for comp = 1:numVars
+    for comp = 1:length(vf)
```
Comment (lbenet) [inline src/calculus.jl:269]: Same suggestion as above...

File: src/calculus.jl [hunk 3]
```diff
@@ -294,10 +293,9 @@ it is evaluated at zero.
 """
 function jacobian!(jac::Array{T,2}, vf::Array{TaylorN{T},1}) where {T<:Number}
     numVars = get_numvars()
-    @assert length(vf) == numVars
-    @assert (numVars, numVars) == size(jac)
+    @assert (length(vf), numVars) == size(jac)
     for comp2 = 1:numVars
-        for comp1 = 1:numVars
+        for comp1 = 1:length(vf)
```
Comment (lbenet) [inline src/calculus.jl:298]: Same suggestion, again

File: src/calculus.jl [hunk 4]
```diff
@@ -306,10 +304,10 @@ end
 function jacobian!(jac::Array{T,2}, vf::Array{TaylorN{T},1},
         vals::Array{T,1}) where {T<:Number}
     numVars = get_numvars()
-    @assert length(vf) == numVars == length(vals)
-    @assert (numVars, numVars) == size(jac)
+    @assert numVars == length(vals)
+    @assert (length(vf), numVars) == size(jac)
     for comp = 1:numVars
-        @inbounds for nv = 1:numVars
+        @inbounds for nv = 1:length(vf)
```
Comment (lbenet) [inline src/calculus.jl:310]: Same suggestion again

**Review by lbenet — COMMENTED** [review]
Thanks for your contribution. I left essentially one suggestion (four times)... Aside from that, LGTM!

**Review by lbenet — APPROVED** [review]
LGTM. Merging!

**Pull request merged**